peer.go

```diff
@@ -554,10 +554,29 @@ func (p *peer) pushMerkleBlockMsg(sha *wire.ShaHash, doneChan, waitChan chan str
 		return err
 	}
 
-	// Generate a merkle block by filtering the requested block according
-	// to the filter for the peer and fetch any matched transactions from
-	// the database.
-	merkle, matchedHashes := bloom.NewMerkleBlock(blk, p.filter)
+	var merkle *wire.MsgMerkleBlock
+	var matchedHashes []*wire.ShaHash
+	if cfg.NoBloom {
+		// Since bloom filtering support is disabled, generate a merkle block
+		// manually. The merkle block will contain all tx hashes from the
+		// original block.
+		header := blk.MsgBlock().Header
+		numTxs := len(blk.Transactions())
+
+		merkle = wire.NewMsgMerkleBlock(&header)
+		merkle.Transactions = uint32(numTxs)
+		for _, tx := range blk.Transactions() {
+			merkle.Hashes = append(merkle.Hashes, tx.Sha())
+		}
+		merkle.Flags = bytes.Repeat([]byte{0x01}, numTxs)
```

Comment:
This is not correct.  That Flags parameter of a merkle block is a bitset.  This should therefore choosing the correct number of bytes based on the number of bits needed to represent each transaction and setting trailing zeros in the final byte for transactions which are not present.